core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityService.scala

```diff
@@ -76,8 +72,10 @@ class EntityService(protected val userInfo: UserInfo, val dataSource: SlickDataS
       }
 
       entityFuture.recover {
-        case _: EntityNotFoundException => RequestComplete(StatusCodes.NotFound, s"${entityType} ${entityName} does not exist in $workspaceName")
-      }
+        case _: EntityNotFoundException =>
+          // could move this error message into EntityNotFoundException and allow it to bubble up
+          throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"${entityType} ${entityName} does not exist in $workspaceName"))
```

Comment:
Why do we throw here when in other places (like deleteEntities and queryEntities) we use RequestComplete?

Reply:
good catch, changed!